nari/ext/act/parser/ability.py

```diff
@@ -9,6 +9,47 @@
 from nari.types.event import Event
 from nari.ext.act.exceptions import ActLineParsingException
 
+try:
+    from nari_act_rust import ability_from_params
+except ImportError:
+    def ability_from_params(timestamp: Timestamp, params: list[str]) -> Ability:
+        """Internal function of ability_from_logline"""
+        source_actor = Actor(*params[0:2])
+        ability = AbilityEvent(*params[2:4])
+        target_actor = Actor(*params[4:6])
+        action_effects = []
+        for i in range(0, 16, 2):
+            index = i + 6
+            action_effects.append(
+                action_effect_from_logline(params[index:index + 2])
+            )
+        actor_update_resources_and_positions(source_actor, params[22:32])
+        actor_update_resources_and_positions(target_actor, params[32:42])
+
+        sequence_id = int(params[42], 16)
+
+        return Ability(
+            timestamp=timestamp,
+            action_effects=action_effects,
+            source_actor=source_actor,
+            target_actor=target_actor,
+            ability=ability,
+            sequence_id=sequence_id,
+        )
```

Comment:
Seriously, prefer a function to alias one of the explicit implementations based on their existing. Way cleaner than a huge try block, especially since we have 4-space indentation like we live in a barn.

Reply:
Changed to what nono suggested